src/main/java/io/github/externschool/planner/dto/TeacherDTO.java

```diff
@@ -0,0 +1,135 @@
+package io.github.externschool.planner.dto;
+
+import io.github.externschool.planner.entity.SchoolSubject;
+import io.github.externschool.planner.entity.profile.Teacher;
+
+import javax.validation.constraints.NotNull;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+public class TeacherDTO {
+
+    private String verificationKey;
+
+    @NotNull
+    private String firstName;
+
+    @NotNull
+    private String patronymicName;
+
+    @NotNull
+    private String lastName;
+
+    @NotNull
+    private String phoneNumber;
+
+    @NotNull
+    private String officer;
+
+    private Set<SchoolSubject> schoolSubjects = new HashSet<>();
+
+    public TeacherDTO(){
+
+    }
+
+    public TeacherDTO(String verificationKey, String firstName, String patronymicName,
+                      String lastName, String phoneNumber, String officer,
+                      Set<SchoolSubject> schoolSubjects) {
+        this.verificationKey = verificationKey;
+        this.firstName = firstName;
+        this.patronymicName = patronymicName;
+        this.lastName = lastName;
+        this.phoneNumber = phoneNumber;
+        this.officer = officer;
+        this.schoolSubjects = schoolSubjects;
+    }
+
+    public TeacherDTO constructTeacher(Teacher teacher){
+        TeacherDTO teacherDTO = new TeacherDTO();
+
+        teacherDTO.setFirstName(teacher.getFirstName());
+        teacherDTO.setLastName(teacher.getLastName());
+        teacherDTO.setOfficer(teacher.getOfficer());
+        teacherDTO.setPatronymicName(teacher.getPatronymicName());
+        teacherDTO.setPhoneNumber(teacher.getPhoneNumber());
+        teacherDTO.setSchoolSubjects(teacher.getSubjectList());
```

Comment:
use teacher.getSubjects() instead of teacher.getSubjectList(), that method was renamed since it returns Set<SchoolSubject> not a List